zokrates_github_resolver/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+//! # GitHub import resolver
+//!
+//! GitHub import resolver allows to import files located in github.com repos.
+//!
+//! To import file from github, use following syntax:
+//! ```zokrates
+//! import "github:user/repo/branch/path/to/file.code"
+//! ```
+//!
+//! For example:
+//! ```zokrates
+//! import "github:eupn/test/master/examples/merkleTree/sha256PathProof3.code" as merkleTreeProof
+//! ```
+//!
+//! Example above imports file `examples/merkleTree/sha256PathProof3.code` located at @eupn's `test`
+//! repository's `master` branch by downloading from URL:
+//! https://raw.githubusercontent.com/eupn/test/master/examples/merkleTree/sha256PathProof3.code
+//!
+
+use reqwest;
+use std::fs::File;
+use std::io::{self, copy, BufReader};
+use std::path::PathBuf;
+use tempfile::NamedTempFile;
+
+/// Prefix for github import source to be distinguished.
+const GITHUB_IMPORT_PREFIX: &str = "github:";
+
+/// Resolves import from the Github.
+/// This importer needs to be provided with location since relative paths could be used inside the
+/// files that are imported from github.
+pub fn resolve(
+    location: &Option<String>,
+    path: &String,
+) -> Result<(BufReader<File>, String, String), io::Error> {
+    if let Some(location) = location {
+        let (root, repo, branch, path) = parse_input_path(&path)?;
+
+        let pb = download_from_github(&root, &repo, &branch, &path).unwrap();
```

Comment:
As you do error handling in `download_from_github`, it would be nice to return the error here instead of panicking